### PR TITLE
Pull request for collectd/grafana

### PR DESCRIPTION
**Pull request by sileht**
Add collectd and grafana container
Move gnocchi file into a sbudirectory
Put uwsgi.ini outside of Dockerfile

### CODE REVIEW

**Comment by sileht** [comment]
This is work in progress.

**Comment by jd** [comment]
@sileht I'd be interested to use that to run a demo next week. Is that working already?

**Comment by sileht** [comment]
No it doesn't yet

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 5118090a859cd48cf39388205ecc287772bc300b

**Comment by sileht** [comment]
The setup works but it need my collectd and gnocchiclient fixes.

File: collectd/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    python-dev \
+	python-pip \
+    libyaml-dev \
+    build-essential \
+    git \
+    stress \
+ && rm -rf /var/lib/apt/lists/*
+
+RUN pip install envtpl
+RUN pip install collectd-gnocchi
```
Comment (jd) [inline collectd/Dockerfile]: I'd run pip once and for all but ok

File: grafana/run-and-configure.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+
+set -x
+set -e
+
+if [ ! "$GF_INSTALL_PLUGINS" ]; then
+    /install-plugin.sh
+fi
+/run.sh &
+sleep 1
```
Comment (jd) [inline grafana/run-and-configure.sh]: sure?

File: collectd/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,18 @@
+FROM ubuntu:xenial
+
+RUN apt-get update -y && apt-get install -y \
+	collectd \
+    python \
+    python-dev \
+	python-pip \
```
Comment (jd) [inline collectd/Dockerfile]: you got massive alignment issue here buddy :p

File: docker-compose.yaml [hunk 4]
```diff
@@ -36,5 +38,36 @@ services:
...
+    environment:
+      # GF_INSTALL_PLUGINS: gnocchixyz-gnocchi-datasource
+      GF_SECURITY_ADMIN_PASSWORD: password
+      GF_GNOCCHI_ENDPOINT: http://gnocchi:8041
+    ports:
+      - "3000:3000"
+    depends_on:
+      - gnocchi-api
+    volumes:
+      - grafana-volume:/var/lib/grafana
```
Comment (jd) [inline docker-compose.yaml]: just curious, why you need a volume?

File: collectd/Dockerfile (hunk 1 above)
Comment (sileht) [inline collectd/Dockerfile]: fixed

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 6d2181af81e225df43d22c407c0adae255448328

File: collectd/Dockerfile (hunk 3 above)
Comment (sileht) [inline collectd/Dockerfile]: fixed

File: docker-compose.yaml (hunk 4 above)
Comment (sileht) [inline docker-compose.yaml]: If you build dashboards you can reuse them between run.

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 5dbf86f3e1b886dd9509c5c09adcc33971102ab5

File: grafana/run-and-configure.sh (hunk 2 above)
Comment (sileht) [inline grafana/run-and-configure.sh]: I'm lazy to wait the grafana tcp port is ready. I can adjust that.

**Comment by sileht** [comment]
Pull-request updated, HEAD is now f487706342affb4907f5a04a5f3b918b5ce5f912

**Comment by jd** [comment]
It fails sometimes:
```
grafana_1          | + set -e
grafana_1          | + GRAFANA_URL=http://admin:password@grafana:3000/api
grafana_1          | + curl -v -H Accept: application/json -H Content-Type: application/json http://admin:password@grafana:3000/api/datasources
grafana_1          | t=2017-07-01T10:07:40+0000 lvl=info msg="Executing migration" logger=migrator id="Add index for dashboard_id in dashboard_tag"
grafana_1          | t=2017-07-01T10:07:40+0000 lvl=info msg="Executing migration" logger=migrator id="Update dashboard table charset"
grafana_1          | t=2017-07-01T10:07:40+0000 lvl=info msg="Executing migration" logger=migrator id="Update dashboard_tag table charset"
grafana_1          | t=2017-07-01T10:07:40+0000 lvl=info msg="Executing migration" logger=migrator id="create data_source table"
grafana_1          | * Hostname was NOT found in DNS cache
grafana_1          |   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
grafana_1          |                                  Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0t=2017-07-01T10:07:40+0000 lvl=info msg="Executing migration" logger=migrator id="add index data_source.account_id"
grafana_1          | *   Trying 172.18.0.7...
grafana_1          | * connect to 172.18.0.7 port 3000 failed: Connection refused
grafana_1          | * Failed to connect to grafana port 3000: Connection refused
grafana_1          | * Closing connection 0
grafana_1          | curl: (7) Failed to connect to grafana port 3000: Connection refused
gnocchidocker_grafana_1 exited with code 7
```

**Comment by sileht** [comment]
Pull-request updated, HEAD is now e579bcd100b3599e3501b9b49c3627e0913036ce

**Comment by sileht** [comment]
I have removed the sleep 1 and adds check that the service is UP instead.

**Comment by jd** [comment]
At least on second boot I always get
```
fake-phy-host_1    | collectd host: fake-phy-host-43ab7b335b41
fake-phy-host_1    | Error: collectd.conf.tpl
```

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 5e53ea2e0340cf00ed6ef90e08d588cbb20ed18d

**Comment by sileht** [comment]
I have fixed that.

**Pull request merged**